### PR TITLE
ibus-bamboo: init at 0.6.6

### PR DESCRIPTION
**Pull request by SuperBo**
Add IBus Bamboo engine for Vietnamese

###### Motivation for this change

Currenty, NixOS doesn't have any support for typing Vietnamese. ibus-unikey and ibus-bamboo are two popular engines for Vietnamese input. However, ibus-bamboo is more updated than the old ibus-unikey.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8821,4 +8821,10 @@
     github = "cpcloud";
     githubId = 417981;
   };
+  superbo = {
```
Comment (alexarice) [inline maintainers/maintainer-list.nix:8824]: The addition to `maintainer-list.nix` should be in a separate commit

Comment (SuperBo) [reply maintainers/maintainer-list.nix]: @alexarice : thank you, I just reformat my commits.

**Comment by SuperBo** [comment]
Any update on this?

**Comment by thongpv87** [comment]
@worldofpeace

**Comment by worldofpeace** [comment]
I came up with the following diff to suggest as changes
```diff
diff --git a/pkgs/tools/inputmethods/ibus-engines/ibus-bamboo/default.nix b/pkgs/tools/inputmethods/ibus-engines/ibus-bamboo/default.nix
index 5992cbd7ca3..062274c99ee 100644
--- a/pkgs/tools/inputmethods/ibus-engines/ibus-bamboo/default.nix
+++ b/pkgs/tools/inputmethods/ibus-engines/ibus-bamboo/default.nix
@@ -1,5 +1,6 @@
 { stdenv
 , fetchFromGitHub
+, fetchpatch
 , gettext
 , xorg
 , pkgconfig
@@ -20,29 +21,35 @@ stdenv.mkDerivation rec {
     sha256 = "0v7n86bpyplfx617xiwwr4kkpjy3w55aqhd6fhljz6q69hgpvcmm";
   };
 
+  patches = [
+    # Add PREFIX to makefile
+    (fetchpatch {
+      url = "https://github.com/BambooEngine/ibus-bamboo/commit/01b9214cbe479e120de3af7d18df6bf575bead2b.patch";
+      sha256 = "1j66hlz008h1pxb9gngq1mldkynlikx5nldvpk3rpdx28ak1lvz9";
+    })
+  ];
+
   nativeBuildInputs = [
     gettext
     pkgconfig
     wrapGAppsHook
     go
-    xorg.libX11
-    xorg.libXtst
   ];
 
   buildInputs = [
-    ibus
-    gtk3
+    xorg.libX11
+    xorg.xorgproto
+    xorg.libXtst
+    xorg.libXi
   ];
 
   preConfigure = ''
     export GOCACHE="$TMPDIR/go-cache"
-    sed -i 's,/usr/lib,/lib,' Makefile
-    sed -i "s,/usr,$out," bamboo.xml
   '';
 
-  installPhase = ''
-    make DESTDIR="$out" engine_dir=/share/ibus-bamboo ibus_dir=/share/ibus install
-  '';
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+  ];
 
   meta = with stdenv.lib; {
     isIbusEngine = true;
```

The main issue was the dependencies, what was in nativeBuildInputs for x11 dependencies were runtime dependencies. What is in buildinputs now is from looking at all the `#include` in source to make sure it's perfect.

The one thing I wasn't sure on was need to remove `ibus` and gtk3`. From looking at the source I don't see it need their headers, or are they needed in the wrapper somehow?

**Comment by SuperBo** [comment]
@worldofpeace: the patch have been included in v0.6.6. However, to make it work, we still need to edit bamboo.xml file. So I updated the package to v0.6.6. Please have check on it.

Thank you

**Review by worldofpeace — APPROVED** [review]
LGTM. Can you squash the update commit into the init commit?

**Comment by SuperBo** [comment]
> LGTM. Can you squash the update commit into the init commit?

@worldofpeace: I squash two commits. Please have a review.

Thank you
